algoliasearch/transport.go

```diff
@@ -95,8 +95,11 @@ func NewTransportWithHosts(appID, apiKey string, hosts []string) *Transport {
 
 // setTimeout changes the timeouts used by the underlying HTTP client.
 func (t *Transport) setTimeout(connectTimeout, readTimeout time.Duration) {
-	t.httpClient.Transport.(*http.Transport).TLSHandshakeTimeout = connectTimeout
-	t.httpClient.Transport.(*http.Transport).ResponseHeaderTimeout = readTimeout
+	switch transport := t.httpClient.Transport.(type) {
+	case *http.Transport:
+		transport.TLSHandshakeTimeout = connectTimeout
+		transport.ResponseHeaderTimeout = readTimeout
+	}
```

Comment:
Seems reasonable, but how would you like me to log out the error message? I could import the stdlib [log package](https://golang.org/pkg/log/) but it doesn't look like this repo currently outputs anything, and I think adding an error return is undesired here.